src/utils/daos_hdlr.c

```diff
@@ -1645,6 +1646,80 @@ fs_copy_file(struct cmd_args_s *ap,
 	return rc;
 }
 
+#define SUB_STR_LEN 256
+static int
+fs_copy_symlink(struct cmd_args_s *ap,
+		struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+		struct stat *src_stat,
+		const char *src_path,
+		const char *dst_path)
+{
+	int		rc;
+	daos_size_t	len = 0;
+	char		symlink_value[PATH_MAX-SUB_STR_LEN];
+
+	len = PATH_MAX - SUB_STR_LEN - 1;
+	if (src_file_dfs->type == POSIX) {
+		len = readlink(src_path, symlink_value, (size_t)len);
+		if (len < 0) {
+			rc = daos_errno2der(errno);
+			DH_PERROR_SYS(ap, rc, "fs_copy_symlink failed on readlink('%s')",
+				      src_path);
+			D_GOTO(out_copy_symlink, rc);
+		}
+	} else if (src_file_dfs->type == DAOS) {
+		rc = dfs_sys_readlink(src_file_dfs->dfs_sys, src_path, symlink_value, &len);
+		if (rc != 0) {
+			rc = daos_errno2der(rc);
+			DH_PERROR_SYS(ap, rc, "fs_copy_symlink failed on dfs_sys_readlink('%s')",
+				      src_path);
+			D_GOTO(out_copy_symlink, rc);
```

Comment:
Assuming dfs_sys_readlink does return a system error number then you could use the following here, alternatively use DH_PERROR_DER after the conversion. 
```suggestion
			DH_PERROR_SYS(ap, rc, "fs_copy_symlink failed on dfs_sys_readlink('%s')",
				      src_path);
			D_GOTO(out_copy_symlink, rc = daos_errno2der(rc));
```

Reply:
thx! I updated the code with the style you recommended.